tools/socktap/link_layer.cpp

```diff
@@ -29,7 +30,43 @@ create_link_layer(boost::asio::io_service& io_service, const EthernetDevice& dev
         namespace ip = boost::asio::ip;
         ip::udp::endpoint multicast(ip::address::from_string("239.118.122.97"), 8947);
         link_layer.reset(new UdpLink { io_service, multicast });
+    } else if (name.substr(0, 3) == "tcp") {
+        namespace ip = boost::asio::ip;
+
+        TcpLink* tcp = new TcpLink { io_service };
+
+        std::string tcp_ip;
+        unsigned short tcp_port, tcp_ip_len;
+
+        if (vm.count("tcp-connect")) {
```

Comment:
I am not sure if `vm.count` is necessary: An empty vector of string would be just fine.